src/abi/map.ts

```diff
@@ -325,6 +325,73 @@ export const MapFunctions: Map<string, AbiFunction> = new Map([
             },
         },
     ],
+    [
+        "del",
+        {
+            name: "del",
+            resolve(ctx, args, ref) {
+                // Check arguments
+                if (args.length !== 2) {
+                    throwError("del expects one argument", ref); // Ignore self argument
+                }
+                const self = args[0];
+                if (!self || self.kind !== "map") {
+                    throwError("del expects a map as self argument", ref); // Should not happen
+                }
+
+                // Check key type
+                if (args[1].kind !== "ref" || args[1].optional) {
+                    throwError(
+                        "del expects a direct type as first argument",
+                        ref,
+                    );
+                }
+                if (args[1].name !== self.key) {
+                    throwError(
+                        `del expects a ${self.key} as first argument`,
+                        ref,
+                    );
+                }
+
+                // Returns nothing
+                return { kind: "void" };
```

Comment:
I think we should let the user know if the key was actually deleted or it was not found in the map, in other words it should have the following signature: `del : map<K,V> -> K -> Bool`, so we return `true` if deletion was successful

Reply:
Done